client/blocks/edit-gravatar/index.jsx

```diff
@@ -153,11 +155,70 @@ export class EditGravatar extends Component {
 	};
 
 	render() {
-		const { isUploading, translate, user, additionalUploadHtml } = this.props;
+		const { isGravatarProfileHidden, isUploading, translate, user, additionalUploadHtml } =
+			this.props;
 		const gravatarLink = `https://gravatar.com/${ user.username || '' }`;
 		// use imgSize = 400 for caching
 		// it's the popular value for large Gravatars in Calypso
 		const GRAVATAR_IMG_SIZE = 400;
+
+		if ( this.props.isFetchingUserSettings ) {
+			return (
+				<div className="edit-gravatar edit_gravatar__is-loading">
+					<div className="edit-gravatar__image-container">
+						<div className="edit-gravatar__gravatar-placeholder"></div>
+					</div>
+					<div>
+						<p className="edit-gravatar__explanation edit-gravatar__explanation-placeholder"></p>
+					</div>
+				</div>
+			);
+		}
+
+		if ( isGravatarProfileHidden ) {
+			return (
```

Comment:
I like to keep the `render` method as slim as possible for edge cases. Would it be cleaner if we extracted those edge-case markups to a method and then call it here?

Reply:
Did with c3281ed5c9